### ISSUE TITLE
read.index() should handle UCAR SOI data

### ISSUE THREAD

**Issue by dankelley** [dankelley/oce#1105]
An example is below. I haven't looked at `read.index` in a while, so a reminder: I will need to ensure that the creation of the `soi` dataset in `ocedata` is not going to be affected by any changes. Also, I'd say that `read.index` needs a new argument for missing values, which are clearly -99.9 in this case but might be pretty much anything and therefore may be hard to detect.  (Maybe `oce` needs a function like `guessMissingValue` that would look for negative numbers consisting of all nines, or something like that.)
```R
> i<-read.index("http://www.cgd.ucar.edu/cas/catalog/climind/SOI.signal.ascii")
Error in read.index("http://www.cgd.ucar.edu/cas/catalog/climind/SOI.signal.ascii") : 
  cannot find missing-value token
> l<-readLines("http://www.cgd.ucar.edu/cas/catalog/climind/SOI.signal.ascii")
> head(l)
[1] " 1866  -1.2  -0.3  -1.0  -0.7   0.1  -0.9  -0.7   0.7  -0.4   0.1   1.6  -0.3"
[2] " 1867   0.4  -0.0  -0.0   0.8   0.7  -0.5   0.6   0.5   0.1  -0.7  -1.2  -1.7"
[3] " 1868  -0.4  -0.8  -2.7   0.2  -1.9  -2.3  -0.8  -2.2  -2.2  -2.0  -2.5   1.0"
[4] " 1869  -3.9  -0.6  -0.8   2.5   2.1   1.6   2.1   1.3   0.2   1.5   0.9   0.6"
[5] " 1870   1.9   0.4  -1.2   0.4  -0.3  -1.2   0.1   1.4  -0.0  -0.9  -1.4  -2.4"
[6] " 1871  -2.4  -0.3  -3.4  -0.5  -0.6   0.2   0.4   0.6   1.0  -0.2  -0.0   1.8"
> tail(l)
[1] " 2011   4.2   4.7   3.8   3.2   0.3  -0.1   1.7   0.2   2.1   1.0   1.9   4.4"
[2] " 2012   1.9   0.3   0.2  -0.9  -0.4  -1.5  -0.4  -1.0   0.4   0.2   0.4  -1.5"
[3] " 2013  -0.4  -1.1   1.9   0.0   1.1   1.5   1.2  -0.3   0.6  -0.5   1.1  -0.1"
[4] " 2014   2.5  -0.6  -2.8   1.0   0.6  -0.4  -0.5  -2.0  -1.3  -1.5  -1.8  -1.4"
[5] " 2015  -1.9  -0.0  -2.4  -0.4  -1.7  -1.7  -2.4  -3.4  -3.1  -3.5  -1.1  -1.4"
[6] " 2016 -99.9 -99.9 -99.9 -99.9 -99.9 -99.9 -99.9 -99.9 -99.9 -99.9 -99.9 -99.9"
```

**Comment by dankelley** [comment]
As usual, one of the first tasks is going to be figuring out a name for this data type, so I can distinguish its handling from the handling of what the docs presently call the "NOAA" type. I think maybe it should be called "NCAR" since that's listed prominently on the base site from which I got it (http://www.cgd.ucar.edu/cas/catalog/climind/soi.html).  But, oh jeeze Louise, so is "UCAR" and "GCD".

It would be nice if website articles had *author* listings, wouldn't it, dear web?

**Comment by dankelley** [comment]
Fixed in `develop` commit 4d3c85c5ff6927f6d2bdf7a0c53cbc5207fd3532
